src/components/views/elements/AccessibleButton.tsx

```diff
@@ -101,22 +100,26 @@ interface RenderedElementProps extends React.InputHTMLAttributes<Element> {
  * as a button.  Identifies the element as a button, setting proper tab
  * indexing and keyboard activation behavior.
  *
+ * If a ref is passed, it will be forwarded to the rendered element as specified using the `element` prop.
```

Comment:
```suggestion
 * If a ref is provided, it will be forwarded to the rendered element (as specified using the `element` prop).
 * If `element` is a DOM element, the ref will be updated to point to the rendered DOM node.
```

Reply:
> If `element` is a DOM element, the ref will be updated to point to the rendered DOM node.

`element` can only be a string though, not a DOM element. It can be a name of a DOM element but it also can't be anything else (other than undefined where the value is `div`) so not sure what this 2nd line is trying to say.

Reply:
My understanding is that `element` can either be an intrinsic DOM element (`div`, `button`, or whatever), or a React component (`MyFancyButton`). (See https://www.typescriptlang.org/docs/handbook/jsx.html#type-checking, etc)

* If it's a react component, we can't really say much about what happens to the ref other than that it is forwarded to that component for it to do with as it chooses. (It might set it to 42, for all we know.)
* However, if it's an intrinsic DOM element, then we can be much more helpful: the act of "forwarding" it means that `ref.content` will be updated to point to the rendered DOM node corresponding to that element. This is what the user of `AccessibleButton` needs to know, and is the reason that I wanted to add that second sentence.

Now: if you're saying that `element` can't in fact be a React component and can only ever be an intrinsic DOM element, then we can clarify this further (since "forwarding" is entirely an implementation detail of AccessibleButton). Maybe something like:


```suggestion
 * If a ref is provided, it will be updated to point to the rendered DOM node (which will normally be a `div`,
 * unless overridden via the `element` property).
```

Reply:
![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/2067cd2d-8d3d-456b-8144-f047d70f88b4)

The T type generic can only be one of these keys:

![image](https://github.com/matrix-org/matrix-react-sdk/assets/2403652/71a651b1-a392-4aa3-85aa-3dfd06478d06)
(continued...)

Reply:
yup I misread the typescript. (#include "I shouldn't have to reverse-engineer typescript types.rant")